src/WallGo/config.py

```diff
@@ -0,0 +1,279 @@
+
+    ## The following parameters are only used for detonation solutions ##
+    vwMaxDeton: float = 0.99
+    """ Maximal Velocity at which the solver will look to find a detonation solution """
+
+    nbrPointsMinDeton: int = 5
+    """ Minimal number of points probed to bracket the detonation roots. """
+
+    nbrPointsMaxDeton: int = 20
+    """ Maximal number of points probed to bracket the detonation roots. """
+
+    overshootProbDeton: float = 0.05
+    """
+    Desired probability of overshooting a root. Must be between 0 and 1. A smaller value
+    will lead to more pressure evaluations (and thus a longer time), but is less likely
+    to miss a root.
+    """
+
+@dataclass
+class ConfigHydrodynamics:
+    """ Holds the config of the Hydrodynamics class. """
+
+    tmin: float = 0.01
+    """ Minimum temperature that is probed in Hydrodynamics (in units of Tnucl). """
+
+    tmax: float = 10.0
+    """ Maximum temperature that is probed in Hydrodynamics (in units of Tnucl). """
+
+    relativeTol: float = 1e-6
+    """ Relative tolerance used in Hydrodynamics. """
+
+    absoluteTol: float = 1e-10
+    """ Absolute tolerance used in Hydrodynamics. """
+
+@dataclass
+class ConfigThermodynamics:
+    """ Holds the config of the Hydrodynamics class. """
+
+    tmin: float = 0.8
+    """
+    Minimum temperature used in the phase tracing (in units of the estimate for the
+    minimum temperature obtained in the template model). 
+    """
+
+    tmax: float = 1.2
+    """
+    Maximum temperature used in the phase tracing (in units of the estimate for the
+    maximum temperature obtained in the template model). 
+    """
+
+    phaseTracerTol: float = 1e-6
+    """
+    Desired accuracy of the phase tracer and the resulting FreeEnergy interpolation.
+    """
+
+@dataclass
+class ConfigBoltzmannSolver:
+    """ Holds the config of the BoltzmannSolver class. """
+
+    basisM: str = 'Cardinal'
+    """ The position polynomial basis type, either 'Cardinal' or 'Chebyshev'. """
+
+    basisN: str = 'Chebyshev'
+    """ The momentum polynomial basis type, either 'Cardinal' or 'Chebyshev'. """
+
+    collisionMultiplier: float = 1.0
+    """
+    Factor multiplying the collision term in the Boltzmann equation. Can be used for
+    testing or for studying the solution's sensibility to the collision integrals. Don't
+    forget to adjust meanFreePathScale accordingly if this is different from 1
+    (meanFreePathScale should scale like 1/collisionMultiplier).
+    WARNING: THIS CHANGES THE COLLISION TERMS WRT TO THEIR PHYSICAL VALUE.
+    """
+
+@dataclass
+class Config:
+    """
+    Data class that holds all the model-independent configs.
+    It contains objects of the data classes ConfigGrid, ConfigEOM,
+    ConfigEffectivePotential, ConfigHydrodynamics, ConfigThermodynamics and
+    ConfigBoltzmannSolver.
+    It can also load the configs from an .ini file.
+    """
+
+    configGrid: ConfigGrid = field(default_factory=lambda: ConfigGrid())

```

Comment:
Ugly compared to having a non-dataclass but with member attributes, but Benoit said this can produce better Sphinx docstrings. So let's go with this for now